curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656707e11e671ed54c96f7ec.md

```diff
@@ -24,7 +24,7 @@ You should replace `pass` with `array[sorted_index] = left_part[left_array_index
     const merge_sort = __helpers.python.getDef("\n" + transformedCode, "merge_sort");
     const { function_body } = merge_sort;
 
-    assert.match(function_body, /while\s+left_array_index\s*<\s*len\(\s*left_part\s*\)\s+and\s+right_array_index\s*<\s*len\(\s*right_part\s*\):\s*[^}]*if\s+left_part\s*\[\s*left_array_index\s*\]\s*<\s*right_part\s*\[\s*right_array_index\s*\]\s*:\s*[^}]*array\s*\[\s*sorted_index\s*\]\s*=\s*left_part\s*\[\s*left_array_index\s*\]\s*(?!.*\bpass\b)/);
+    assert.match(function_body, /while\s+left_array_index\s*<\s*len\(\s*left_part\s*\)\s+and\s+right_array_index\s*<\s*len\(\s*right_part\s*\):\s*[^}]*if\s+left_part\s*\[\s*left_array_index\s*\]\s*<\s*right_part\s*\[\s*right_array_index\s*\]\s*:\s*((?!pass).)*array\s*\[\s*sorted_index\s*\]\s*=\s*left_part\s*\[\s*left_array_index\s*\]\s*(?!.*\bpass\b)/);
   }
```

Comment:
Hey, we are trying to switch from regex helpers to the new ast-based helpers. It would be great if you could use those. With the ast-based helpers this test would be:
```js
({ test: () => assert(runPython(`
  _Node(_code).find_function("merge_sort")
  .find_whiles()[0].find_ifs()[0].find_bodies()[0]
  .is_equivalent("array[sorted_index] = left_part[left_array_index]")
`)) })
```
Let me know if you want to have a try. [Here](https://opensource.freecodecamp.org/curriculum-helpers/python.html#ast-based-helpers) you can read more. Feel free to ask in case of any doubts.

Reply:
I'm definitely giving it a try, gonna push a commit with ast helpers soon